internal/route/always_route_filtering_strategy.go

```diff
@@ -0,0 +1,133 @@
+package route
+
+import (
+	"reflect"
+	"strings"
+
+	"github.com/satsuma-data/node-gateway/internal/config"
+	"github.com/satsuma-data/node-gateway/internal/metadata"
+	"github.com/satsuma-data/node-gateway/internal/types"
+	"go.uber.org/zap"
+)
+
+type AlwaysRouteFilteringStrategy struct {
+	BackingStrategy  RoutingStrategy
+	Logger           *zap.Logger
+	NodeFilters      []NodeFilter
+	RemovableFilters []NodeFilterType
+}
+
+func (s *AlwaysRouteFilteringStrategy) RouteNextRequest(
+	upstreamsByPriority types.PriorityToUpstreamsMap,
+	requestMetadata metadata.RequestMetadata,
+) (string, error) {
+	// Create a copy of removable filters to avoid modifying the original slice.
+	removableFilters := make([]NodeFilterType, len(s.RemovableFilters))
+	copy(removableFilters, s.RemovableFilters)
+
+	filters := s.NodeFilters // WARNING: Slice assignment, not a copy!
+	for len(filters) > 0 {
+		// Get all healthy upstreams for the current set of filters.
+		upstreams := s.FilterUpstreams(upstreamsByPriority, requestMetadata, filters)
+
+		// If there is at least one healthy upstream, route using the backing strategy.
+		if len(upstreams) > 0 {
+			return s.BackingStrategy.RouteNextRequest(upstreams, requestMetadata)
+		}
+
+		// There are no more filters to remove and no healthy upstreams found, so give up.
+		if len(removableFilters) == 0 {
+			break
+		}
+
+		// Remove the next filter and try again.
+		idx := len(removableFilters) - 1
+		nextFilterToRemove := removableFilters[idx]
+		removableFilters = removableFilters[:idx]
+		filters = removeFilters(filters, nextFilterToRemove)
```

Comment:
can we add a debug log on the request, what filter we're removing, and the remaining filters?